Sources/Private/Model/DictionaryInitializable.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  DictionaryInitializable.swift
+//  Lottie
+//
+//  Created by Marcelo Fabri on 5/5/22.
+//
+
+import Foundation
+
+// MARK: - InitializableError
+
+enum InitializableError: Error {
+  case invalidInput
+}
+
+// MARK: - DictionaryInitializable
+
+protocol DictionaryInitializable {
+
+  init(dictionary: [String: Any]) throws
+
+}
+
+// MARK: - AnyInitializable
+
+protocol AnyInitializable {
+
+  init(value: Any) throws
+
+}
+
+extension Dictionary {
+
+  @_disfavoredOverload
+  func valueFor<T>(key: Key) throws -> T {
```

Comment:
Why are both of these methods necessary?

Reply:
There are cases where we don't implement `AnyInitializable` (eg primitives like `Double`) so we can't simply add the constraint.

I couldn't find a good way to dynamically check whether `T` conforms to `AnyInitializable`, so had to create a separate function

Reply:
Makes sense. Is this `@_disfavoredOverload` annotation necessary? My understanding is that Swift calls the most-specific method that applies to a given type, so types that conform to `AnyInitializable` would automatically use the other method (without needing `@_disfavoredOverload`).

Reply:
Yeah, without it the test was failing since the version with the constraint was never called 😅